feature/security/gnsi/acctz/tests/record_subscribe_partial/record_subscribe_partial_test.go

```diff
@@ -0,0 +1,284 @@
+	if err != nil {
+		t.Fatalf("Failed getting accountz record subscribe client, error: %s", err)
+	}
+
+	err = acctzSubClient.Send(&acctzpb.RecordRequest{
+		Timestamp: firstResponse.Timestamp,
+	})
+	if err != nil {
+		t.Fatalf("Failed sending accountz record request, error: %s", err)
+	}
+
+	secondResponse, err := acctzSubClient.Recv()
+	if err != nil {
+		t.Fatalf("Failed receiving second record response, error: %s", err)
+	}
+
+	if reflect.DeepEqual(firstResponse, secondResponse) {
+		t.Fatalf("Accountz server responded with same event on subsequent record request.")
+	}
+
+	var recordIdx int
+
+	var lastTimestampUnixMillis int64
+
+	for {
+		if recordIdx >= len(records) {
+			t.Log("Out of records to process...")
+			break
+		}
+
+		r := make(chan recordRequestResult)
+
+		go func(r chan recordRequestResult) {
+			var response *acctzpb.RecordResponse
+			response, err = acctzSubClient.Recv()
+			r <- recordRequestResult{
+				record: response,
+				err:    err,
+			}
+		}(r)
+
+		var done bool
+		var resp recordRequestResult
+
+		select {
+		case rr := <-r:
+			resp = rr
+		case <-time.After(10 * time.Second):
+			done = true
+		}
+
+		if done {
+			t.Log("Done receiving records...")
+			break
+		}
+
+		if resp.err != nil {
+			t.Fatalf("Failed receiving record response, error: %s", resp.err)
+		}
+
+		if resp.record.GetHistoryIstruncated() {
+			t.Fatalf("History is truncated but it shouldn't be, Record Details: %s", prettyPrint(resp.record))
+		}
+
+		if !resp.record.Timestamp.AsTime().After(startTime) {
+			// Skipping record, was before test start time.
+			continue
+		}
+
+		// Check that the timestamp for the record is between our start/stop times for our rpc.
+		timestamp := resp.record.Timestamp.AsTime()
+		if timestamp.UnixMilli() == lastTimestampUnixMillis {
+			// This ensures that timestamps are actually changing for each record.
+			t.Fatalf("Timestamp is the same as the previous timestamp, this shouldn't be possible!, Record Details: %s", prettyPrint(resp.record))
+		}
+		lastTimestampUnixMillis = timestamp.UnixMilli()
+
+		// -2 for a little breathing room since things may not be perfectly synced up time-wise.
+		if records[recordIdx].StartTime.Unix()-2 > timestamp.Unix() {
+			t.Fatalf(
+				"Record timestamp is prior to rpc start time timestamp, rpc start timestamp %d, record timestamp %d, Record Details: %s",
+				records[recordIdx].StartTime.Unix()-2,
+				timestamp.Unix(),
+				prettyPrint(resp.record),
+			)
+		}
+
+		// Done time (that we recorded when making the rpc) + 2 second for some breathing room.
+		if records[recordIdx].DoneTime.Unix()+2 < timestamp.Unix() {
+			t.Fatalf(
+				"Record timestamp is after rpc end timestamp, rpc end timestamp %d, record timestamp %d, Record Details: %s",
+				records[recordIdx].DoneTime.Unix()+2,
+				timestamp.Unix(),
+				prettyPrint(resp.record),
+			)
+		}
+
+		serviceType := resp.record.GetGrpcService().GetServiceType()
+		if records[recordIdx].RPCType != serviceType {
```

Comment:
I think you can not assign serviceType, just query it in the if statement.

(similarly for all of the  other assign/access patterns here.

Reply:
there is an argument here for 'readability of a 150 char long line...

Reply:
Replaced with `cmp.Diff()`